hack/check-olm.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+FAILED="false"
+
+# clone OLM
+mkdir olm-test
+cd olm-test
+git clone https://github.com/operator-framework/operator-lifecycle-manager.git
+cd operator-lifecycle-manager
```

Comment:
I'm not a huge fan of the cloning but the alternatives seem more complicated for no real value. So okay to cloning.

Reply:
The only alternative that I think could be better is to get a list of the releases from the GitHub API by making a request to:
https://api.github.com/repos/operator-framework/operator-lifecycle-manager/releases and then parsing those values out.

After doing some digging this could be the best way because then it doesn't include the git tags that aren't associated with an actual release. And it gets rid of the need to clone the repo :smile: . 

I will hold this PR and look into doing this instead as it would definitely be a better method.